infer/resource.go

```diff
@@ -905,11 +909,32 @@ func (rc *derivedResourceController[R, I, O]) Create(ctx p.Context, req p.Create
 	}
 
 	id, o, err := (*r).Create(ctx, req.Urn.Name(), input, req.Preview)
-	if err != nil {
+	if initFailed := (ResourceInitFailedError{}); errors.As(err, &initFailed) {
+		defer func(createErr error) {
```

Comment:
Why are you deferring the error handling here? Does all the following code allow for a failed create? Is a ResourceInitFailedError Create guaranteed to return a valid id and state?

Reply:
> Why are you deferring the error handling here? Does all the following code allow for a failed create?

If we get a `ResourceInitFailedError`, we still want to serialize the output and return, but we also want to keep the context from the original error. It does allowed for a fully failed create, in which case the error is just returned.

> Is a ResourceInitFailedError Create guaranteed to return a valid id and state?

A correct implementation will return a valid enough id and state to be handled in by the resource's `Update` method.